### DISCUSSION TITLE
How to install GDAL/OGR with GeoParquet on OS/X

### DISCUSSION THREAD

**Discussion by cholmes** [opengeospatial/geoparquet#99] [Ecosystem]
I'm trying to get going with GDAL 3.5.0, but I can't seem to get the GeoParquet driver showing up.

I managed to get GDAL installed with homebrew, and it shows up as version 3.5.0.

But I do:
```
 % ogrinfo ~/Repos/geoparquet/examples/example.parquet 
FAILURE:
Unable to open datasource `/Users/cholmes/Repos/geoparquet/examples/example.parquet' with the following drivers.
  -> ESRIC
  -> FITS
  -> PCIDSK
  -> netCDF
  -> PDS4
  -> VICAR
  -> JP2OpenJPEG
  -> PDF
  -> MBTiles
  -> BAG
  -> EEDA
  -> OGCAPI
  -> ESRI Shapefile
  -> MapInfo File
  -> UK .NTF
  -> LVBAG
  -> OGR_SDTS
  -> S57
  -> DGN
  -> OGR_VRT
  -> Memory
  -> CSV
  -> NAS
  -> GML
  -> GPX
  -> KML
  -> GeoJSON
  -> GeoJSONSeq
  -> ESRIJSON
  -> TopoJSON
  -> Interlis 1
  -> Interlis 2
  -> OGR_GMT
  -> GPKG
  -> SQLite
  -> ODBC
  -> WAsP
  -> PGeo
  -> MSSQLSpatial
  -> PostgreSQL
  -> OpenFileGDB
  -> DXF
  -> CAD
  -> FlatGeobuf
  -> Geoconcept
  -> GeoRSS
  -> VFK
  -> PGDUMP
  -> OSM
  -> GPSBabel
  -> OGR_PDS
  -> WFS
  -> OAPIF
  -> EDIGEO
  -> SVG
  -> Idrisi
  -> XLS
  -> ODS
  -> XLSX
  -> Elasticsearch
  -> Carto
  -> AmigoCloud
  -> SXF
  -> Selafin
  -> JML
  -> PLSCENES
  -> CSW
  -> VDV
  -> GMLAS
  -> MVT
  -> NGW
  -> MapML
  -> TIGER
  -> AVCBin
  -> AVCE00
  -> HTTP
```

Doesn't seem to be listed. I see that there's a 'build dependency' of 'Parquet component of the Apache Arrow C++ library'. So I tried a `brew install apache-arrow` to see if that might help and it didn't. Tried to reinstall gdal with apache-arrow there, but that didn't help. 

So not sure what to try next. I was expecting to have it automatically there with `brew install`, but is there some extra step I need to do?

**Comment by kylebarron** [comment]
It does seem like Parquet is not included in the default build config. In the Docker image it's there
```
> docker run -it osgeo/gdal:latest ogrinfo --formats | grep Parquet
  Parquet -vector- (rw+v): (Geo)Parquet
```
but with local homebrew it's not. 
```
> ogrinfo --formats | grep Parquet
```

It looks like the [homebrew build script](https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/gdal.rb) for GDAL would need to be updated.

**Reply by rouault** [reply]
> It looks like the [homebrew build script](https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/gdal.rb) for GDAL would need to be updated.

It would need a full rewrite to switch to CMake, since that's required for the Parquet driver, and anyway autoconf support will be removed to GDAL 3.6.0. I've filed an issue in https://github.com/Homebrew/homebrew-core/issues/101771 , but not sure if that was the right place

**Reply by rouault** [reply]
hum as I suspected my issue got closed as the only accepted contributions are pull requests... Not something practical to do for me for Homebrew

**Reply by cholmes** [reply]
Who did the original gdal in homebrew work? 

Could this be a use of GDAL sponsor funds? Seems like a necessary step in the move to cmake. I'd guess conda, apt, vcpkg, etc need the same thing?

**Reply by rouault** [reply]
> Who did the original gdal in homebrew work?

No idea. Anyway original authors might no longer be involved and https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/gdal.rb shows a lot of people involved, touching small bits there and there

> Could this be a use of GDAL sponsor funds?

I'm not sure. Those are different projects/efforts with their own community and skills. Probably a discussion for another forum.

**Reply by cholmes** [reply]
> Probably a discussion for another forum.

Cool. Let me know if that discussion happens anywhere, I'd be interested to follow it and possibly sound in.

**Reply by cholmes** [reply]
Homebrew now works! `brew install gdal` will get GDAL 3.6.0 with geoparquet support built int!

**Reply by jorisvandenbossche** [reply]
BTW, also the conda gdal package supports this since a few days, and it is distributed separately (so you still need to install a separate `libgdal-arrow-parquet`): https://twitter.com/EvenRouault/status/1601607372538904576

**Comment by jorisvandenbossche** [comment]
Since the Parquet driver is opt-in (requiring additional dependencies), that basically means that any distribution of GDAL (brew, conda, apt, vcpkg, ..) will need to be updated explicitly to include Parquet support. 
(so in general I wouldn't expect this already be the case right after the release)

**Reply by kylebarron** [reply]
Maybe we should make a note that the Docker build of gdal includes Parquet, in case people are looking for a way to test it but don't want to compile gdal from scratch.

**Reply by cholmes** [reply]
+1. Yeah, I was hoping to write up some guidance for how people can try it out. Makes sense that things need to be updated, I was just trying my default update path and seeing what happened.

Does anyone know the process of getting those updated and/or the owners? Like can we help with PR's, or at least raise issues to let the owners know that there's a new open format that has additional dependencies?
